test_util/provide_aws.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python3
+import logging
+import os
+from abc import ABCMeta, abstractmethod
+
+import boto3
+import botocore
+import retrying
+
+
+LOGGING_FORMAT = '[%(asctime)s|%(name)s|%(levelname)s]: %(message)s'
+logging.basicConfig(format=LOGGING_FORMAT, level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+AWS_VERIFY = True
+
+
+class AwsProvision(metaclass=ABCMeta):
```

Comment:
what's the value of having this and CloudFormation be separate classes? They both would only work / make sense as currently written for CloudFormation stacks (things like self.stack, self.stack_name, only make sense for CloudFormation)